internal/services/instance_types_service.go

```diff
@@ -61,8 +63,33 @@ func ListInstanceTypes(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := render.RenderList(w, r, payloads.NewListInstanceTypeResponse(&res)); err != nil {
+	instances := convertAWSTypes(res)
+	if err != nil {
+		renderError(w, r, payloads.NewAWSError(r.Context(), "can't convertAWSTypes", err))
+		return
+	}
+
+	filteredInstances, err := supported_types.FilterInstanceTypes(r.Context(), &instances)
+	if err != nil {
+		renderError(w, r, payloads.NewAWSError(r.Context(), "can't FilterUnsupportedTypes", err))
+		return
+	}
+	if err := render.RenderList(w, r, payloads.NewListInstanceTypeResponse(filteredInstances)); err != nil {
 		renderError(w, r, payloads.NewRenderError(r.Context(), "list instance types", err))
 		return
 	}
 }
+
+func convertAWSTypes(types []types.InstanceTypeInfo) []clients.InstanceType {
+	list := make([]clients.InstanceType, 0, len(types))
+	for i := range types {
+		list = append(list, clients.InstanceType{
+			Name:   string(types[i].InstanceType),
+			VCPUs:  types[i].VCpuInfo.DefaultVCpus,
+			Cores:  types[i].VCpuInfo.DefaultCores,
+			Memory: *types[i].MemoryInfo.SizeInMiB,
+		})
+	}
+	return list
+
+}
```

Comment:
Extra empty line.

I think that we have settled on the type and that it should live in the client package, I do believe this conversion function should be at the same place too. This is a service package, but we might need the conversion maybe in other places too.

It should be also exported, one idea for a name would be (trying to mimic Idiomatic Go not sure about it): `NewInstanceType`.